File: tools/claude-app-smoke-fixture.cs
using System;
using System.Linq;
using System.Threading;

public static class ClaudeAppSmokeFixture
{
    public static int Main(string[] args)
    {
        bool proxy = args.Any(value => value.StartsWith("--proxy-server=http://127.0.0.1:", StringComparison.Ordinal));
        bool certificate = args.Any(value => value.StartsWith("--ignore-certificate-errors-spki-list=", StringComparison.Ordinal));
        bool userData = args.Any(value => value.StartsWith("--user-data-dir=", StringComparison.Ordinal));
        if (!proxy || !certificate || !userData)
        {
            return 64;
        }
        Thread.Sleep(5000);
        return 0;
    }
}

File: tools/claude-app-smoke-fixture.c
#define _POSIX_C_SOURCE 199309L

#include <string.h>
#include <time.h>

int main(int argc, char **argv) {
    int proxy = 0;
    int certificate = 0;
    int user_data = 0;
    for (int index = 1; index < argc; index++) {
        proxy |= strncmp(argv[index], "--proxy-server=http://127.0.0.1:",
                         sizeof("--proxy-server=http://127.0.0.1:") - 1) == 0;
        certificate |= strncmp(argv[index], "--ignore-certificate-errors-spki-list=",
                               sizeof("--ignore-certificate-errors-spki-list=") - 1) == 0;
        user_data |= strncmp(argv[index], "--user-data-dir=",
                             sizeof("--user-data-dir=") - 1) == 0;
    }
    if (!proxy || !certificate || !user_data) {
        return 64;
    }
    struct timespec delay = {5, 0};
    nanosleep(&delay, 0);
    return 0;
}
